src/serialize/row.rs

```diff
@@ -0,0 +1,117 @@
+use crate::serialize::value::{
+    PyAnyWrapper, PythonDriverSerializationError, mk_typck_err_named_row, mk_typck_err_row,
+};
+use pyo3::exceptions::PyTypeError;
+use pyo3::prelude::*;
+use pyo3::types::{PyDict, PyList, PyTuple};
+use pyo3::{Bound, Py, PyAny, Python};
+use scylla_cql::_macro_internal::{
+    ColumnSpec, RowSerializationContext, RowWriter, SerializationError, SerializeRow,
+    SerializeValue,
+};
+use std::ops::Deref;
+
+pub(crate) struct PyAnyWrapperRow(pub(crate) Py<PyAny>);
+
+impl Deref for PyAnyWrapperRow {
+    type Target = Py<PyAny>;
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+
+impl PyAnyWrapperRow {
+    fn length_equality_check(
+        row_len: usize,
+        cols_len: usize,
+        name: &'static str,
+    ) -> Result<(), SerializationError> {
+        if row_len != cols_len {
+            Err(SerializationError::new(mk_typck_err_named_row(
+                name,
+                scylla::serialize::row::BuiltinTypeCheckErrorKind::WrongColumnCount {
+                    rust_cols: row_len,
+                    cql_cols: cols_len,
+                },
+            )))
+        } else {
+            Ok(())
+        }
+    }
+
+    fn serialize_element(
+        col: &ColumnSpec,
+        val: Bound<PyAny>,
+        row_writer: &mut RowWriter,
+    ) -> Result<(), SerializationError> {
+        let wrapper = PyAnyWrapper(val);
```

Comment:
❓ Does `PyAnyWrapper` need ownership over `Bound<PyAny>`? Perhaps a reference would be enough?
Let's not make methods take non-`Copy` data by move if borrowing suffice.